core/src/cluster_info_vote_listener.rs

```diff
@@ -330,6 +330,7 @@ impl ClusterInfoVoteListener {
         let mut latest_vote_slot_per_validator = HashMap::new();
         let mut last_process_root = Instant::now();
         let duplicate_confirmed_slot_sender = Some(duplicate_confirmed_slot_sender);
+        let subscriptions = subscriptions.and_then(|arc| Arc::try_unwrap(arc).ok());
```

Comment:
If you want to convert from `Option<Arc<T>>` to `Option<&T>`, it's better to use `as_deref()` https://doc.rust-lang.org/std/option/enum.Option.html#method.as_deref

This means changing the `&Option<T>` to `Option<&T>` and using `as_deref()` in all call sites, or you can use `&Option<&T>`, but I don't think there's any benefit to that.

`try_unwrap()` will fail if there's other references to the RpcSubscriptions, so we should avoid it: https://doc.rust-lang.org/std/sync/struct.Arc.html#method.try_unwrap